encode_test.go

```diff
@@ -521,17 +521,12 @@ var marshalTests = []struct {
 }
 
 func TestMarshal(t *testing.T) {
-	defer os.Setenv("TZ", os.Getenv("TZ"))
-	os.Setenv("TZ", "UTC")
+	t.Setenv("TZ", "UTC")
```

Comment:
t.SetEnv is only available since Go 1.17

So here this change might break things. But only the test. I would keep the old code

Reply:
I reverted the change but I have a bad feeling about supporting versions too low. Do you know where I can learn about the issues preventing some people from using a supported version of go?